### PR TITLE
GLEN-216: Backport migration to FreeRDP 2.0.0.

### PR DESCRIPTION
**Pull request by mike-jumper**
This backporting effort is significantly more complex than others due to the size of the changes involved and the way those changes have been built on the various other changes made since 0.9.12-incubating.

To manage to backport things successfully, this set of changes takes the following approach:

1. Backport features that would both be beneficial and would reduce the amount of manual changes required to make the FreeRDP 2.0.0 changes compatible:
   * Support for configuring the SFTP root directory (#190)
   * Support for configuring SSH host key checking (#191)
2. Backport known dependencies through the usual "cherry-pick" method:
   * Migration to the new unit test tooling (#193)
   * Additional string utility functions within libguac (#194).
3. Copy over the RDP support from the `glyptodon/2.x` branch **verbatim**.
4. Make as minimal changes as possible to bring that support back into compatibility with `glyptodon/1.x`:
   * Use the common session recording API that was part of 0.9.12 guacamole-server (no recording of input events)
   * Remove trace-level logging (not part of 0.9.12 libguac)
   * Remove use of Guacamole protocol error codes that were not part of the 0.9.12 version of the protocol
   * Correct calls to the common SSH utility functions.
   * Remove reliance on the user timezone handshake (not part of the Guacamole protocol until way after 0.9.12)
   * Remove any settings not already part of `glyptodon/1.x`, #190, or #191.

With these changes made, `glyptodon/1.x` builds against FreeRDP 2.0.0 just as `glyptodon/2.x` but remains completely compatible.

### CODE REVIEW

**Comment by mike-jumper** [comment]
> 

So ... effectively only 716 lines! Tiny diff!

**Comment by jmuehlner** [comment]
😨

File: src/protocols/rdp/settings.c [hunk 1]
```diff
@@ -860,7 +860,7 @@ guac_rdp_settings* guac_rdp_parse_args(guac_user* user,
     /* Timezone if provided by client, or use handshake version */
```
Comment (jmuehlner) [inline src/protocols/rdp/settings.c:747]: Is this comment about the handshake version wrong now?

Comment (mike-jumper) [reply src/protocols/rdp/settings.c:747]: Yeeeessss ... I'll correct that as part of the relevant commit.

**Pull request merged**